stats/opencensus/trace.go

```diff
@@ -61,28 +60,9 @@ func (csh *clientStatsHandler) traceTagRPC(ctx context.Context, rti *stats.RPCTa
 // if present.
 func (ssh *serverStatsHandler) traceTagRPC(ctx context.Context, rti *stats.RPCTagInfo) (context.Context, *traceInfo) {
 	mn := strings.Replace(removeLeadingSlash(rti.FullMethodName), "/", ".", -1)
-	var sc trace.SpanContext
-	// gotSpanContext represents if after all the logic to get span context out
-	// of context passed in, whether there was a span context which represents
-	// the future created server span's parent or not.
-	var gotSpanContext bool
-	md, ok := metadata.FromIncomingContext(ctx)
-	if !ok {
-		// Shouldn't happen, but if does just notify caller through error log,
-		// and create a span without remote parent.
-		logger.Error("server side context passed into stats handler has no metadata")
-	} else {
-		tcMDVal := md[traceContextMDKey]
-		if len(tcMDVal) != 0 {
-			tcBin := []byte(tcMDVal[0])
-			// Only if this function gets through all of these operations and the
-			// binary metadata value successfully unmarshals means the server
-			// received a span context, meaning this created server span should have
-			// a remote parent corresponding to whatever was received off the wire
-			// and unmarshaled into this span context.
-			sc, gotSpanContext = propagation.FromBinary(tcBin)
-		}
-	}
+
+	tcBin := stats.Trace(ctx)
+	sc, gotSpanContext := propagation.FromBinary(tcBin)
 
 	var span *trace.Span
 	if gotSpanContext {
```

Comment:
How about:

```go
var span *trace.Span
if sc, ok := propagation.FromBinary(stats.Trace(ctx)); ok {
}
```

(Or keep `tcBin` if desired, but rename `gotSpanContext` to `ok` and move that call into the `if`.)

Reply:
Yeah, I was debating whether or not to read tcBIn into a local var. I decided to for readability (at the cost of a pointless local var if you inline call). I figured you'd say something if you had an opinion on it. Yeah, will switch to ok, keep it consistent with rest of codebase. Thanks for suggestion.